dateparser/date_parser.py

```diff
@@ -63,7 +63,23 @@ def parse(self, timestr, default=None, ignoretz=False, **kwargs):
             if not getattr(res, e):
                 new_date = new_date.replace(**{e: 0})
 
-        return new_date
+        if return_period:
+            return new_date, self.get_period(res)
+        else:
+            return new_date
```

Comment:
I believe this is our internal class, and users should only use `dateparser.parse` or `dateparser.DateDataParser`